### DISCUSSION TITLE
Automerge not working, Branches marked as modified.

### DISCUSSION THREAD

**Discussion by ssi-ips** [renovatebot/renovate#30823] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github Action v40.2.6 (Renovate 38.37.1)

### Please tell us more about your question or problem

I noticed that in one repository the automerge is not working correctly. The last merges are from 3 weeks ago, but the repository doesn't have a lot of dependencies so I can't say for sure.

![grafik](https://github.com/user-attachments/assets/2758745c-5dac-4259-b400-ffc299c98a32)

![grafik](https://github.com/user-attachments/assets/3227448d-4628-4726-be8a-f098593005db)

The only strange thing I noticed is that the PR is marked as edited/blocked. The branch is marked as modified despite no changes besides the version bump by renovate. Forcing a rebase removed the notification but it's added again in the second renovate run.

When setting ignoreTests to true and forcing a rebase the PR is merged. So I figured it has to do with the checks but I can't find the cause, everything looks fine. I even tried approving the PR. 

I hope anyone with more experience in understanding the Logs can help.

Example branch: renovate/com.vaadin-24.x 

### Logs (if relevant)

<details><summary>Logs</summary>

https://gist.github.com/ssi-ips/ab0b8e1b41886768f31f414a925c69d3

</details>

**Comment by rarkins** [comment]
Look for a log message which should print the expected vs existing git author

**Reply by ssi-ips** [reply]
Can't find it. gitAuthor is only mentioned when the config is logged.

**Reply by ssi-ips** [reply]
After finding [this discussion](https://github.com/renovatebot/renovate/discussions/30741) the issue for the edited/blocked notification seems that the author email changed.

I think I found what you meant when enabling trace:

> 2024-08-16T11:20:38.2760833Z TRACE: branch.isModified() = true (repository=Nexus-IPS/driverstore, baseBranch=main, branch=renovate/com.vaadin-24.x)
2024-08-16T11:20:38.2761020Z        "branchName": "renovate/com.vaadin-24.x",
2024-08-16T11:20:38.2761385Z        "committedAuthors": ["122013515+renovate-nexus-ips[bot]@users.noreply.github.com"],
2024-08-16T11:20:38.2761746Z        "includedAuthors": ["122013515+renovate-nexus-ips[bot]@users.noreply.github.com"],
2024-08-16T11:20:38.2761899Z        "gitAuthorEmail": "bot@renovateapp.com",
2024-08-16T11:20:38.2762043Z        "ignoredAuthors": ["bot@renovateapp.com"]
